### PR TITLE
Posix: Add support for POSIX fsync and ftruncate

### PR DESCRIPTION
**Pull request by karthi012**
Adding support for POSIX API for fsync and ftruncate along with testsuite

### CODE REVIEW

**Review by cfriedt — APPROVED** [review]
LGTM - thanks 👍

**Review by ycsin — APPROVED** [review]

File: lib/posix/options/fs.c [hunk 1]
```diff
@@ -412,3 +416,18 @@ int mkdir(const char *path, mode_t mode)
...
+ */
+int ftruncate(int fd, off_t length)
+{
+	struct posix_fs_desc *ptr = NULL;
+
+	ptr = z_get_fd_obj(fd, NULL, EBADF);
+	if (!ptr)
+		return -1;
+
+	return fs_truncate(&ptr->file, length);
```
Comment (kartben) [inline lib/posix/options/fs.c:432]: AFAIK fs_truncate may return other values than 0 or -1, so I think this should be handled properly by setting errno appropriately and returning -1?

Comment (cfriedt) [reply lib/posix/options/fs.c:432]: Good catch @kartben. @karthi012 - can you please address this in a follow-up pr? Also, please ensure that fsync is marked as supported in the docs.

Comment (karthi012) [reply lib/posix/options/fs.c:432]: @kartben Thanks for comments.  @cfriedt, i will address.

**Pull request merged**